OptimizelySDK/Optimizely.cs

```diff
@@ -429,7 +429,8 @@ public virtual bool IsFeatureEnabled(string featureKey, string userId, UserAttri
 
             if (decision != null)
             {
-                var experiment = Config.GetExperimentFromId(decision.ExperimentId);
+                var experiment = decision.Source == FeatureDecision.DECISION_SOURCE_EXPERIMENT ?
```

Comment:
Let's have the `Decision` object carry the experiment instead of just the `experimentId`. That way we can avoid doing an unnecessary lookup.

Reply:
can we do it in separate PR? There are so many unit tests will need to be updated.

Reply:
If we do it in another PR, then this piece of code won't be necessary. I'd rather us just change it here and make that the fix to the bug.

Reply:
Sure mike.